### PR TITLE
Small update to improve perf and memory usage of blockchain cache

### PR DESCRIPTION
**Pull request by NicolasDP**
This is a rather simple change but might break a couple of pending work in different PR.

also contains a fix that should have been in #703

### CODE REVIEW

File: jormungandr/src/blockchain/process.rs [hunk 1]
```diff
@@ -70,11 +70,13 @@ pub fn handle_input(
                 });
 
             if let Some(msg_box) = explorer_msg_box {
+                /*
                 msg_box
                     .try_send(ExplorerMsg::NewBlock(new_block_ref))
                     .unwrap_or_else(|err| {
                         error!(info.logger(), "cannot add block to explorer: {}", err)
                     });
+                */
```
Comment (mzabaluev) [inline jormungandr/src/blockchain/process.rs]: Will this be re-enabled in a follow-up change?

File: jormungandr/src/blockchain/reference_cache.rs [hunk 2]
```diff
@@ -121,11 +121,15 @@ impl RefCacheData {
         }
     }
 
-    fn poll_purge(&mut self) -> Poll<(), timer::Error> {
-        while let Some(entry) = try_ready!(self.expirations.poll()) {
-            self.entries.remove(entry.get_ref());
+    pub fn poll_purge(&mut self) -> Poll<(), timer::Error> {
+        loop {
+            match self.expirations.poll()? {
+                Async::NotReady => return Ok(Async::Ready(())),
```
Comment (mzabaluev) [inline jormungandr/src/blockchain/reference_cache.rs]: It looks like it does not makes sense for this to be a poll, but with std futures, there will be an explicit context to pass down, so OK.

File: jormungandr/src/rest/v0/handlers.rs [hunk 3]
```diff
@@ -19,14 +19,15 @@ use bytes::{Bytes, IntoBuf};
...
+use std::sync::Arc;
 
 pub use crate::rest::Context;
 
 macro_rules! ActixFuture {
     () => { impl Future<Item = impl Responder + 'static, Error = impl Into<Error> + 'static> + 'static }
 }
 
-fn chain_tip_fut<'a>(context: &State<Context>) -> impl Future<Item = Ref, Error = Error> {
+fn chain_tip_fut<'a>(context: &State<Context>) -> impl Future<Item = Arc<Ref>, Error = Error> {
```
Comment (mzabaluev) [inline jormungandr/src/rest/v0/handlers.rs]: Maybe `Ref` could encapsulate the `Arc` under the hood, but maybe explicit is better.

Comment (NicolasDP) [reply jormungandr/src/rest/v0/handlers.rs]: sure that would make sense. there are pro and cons to encapsulate and hide the `Arc`.

File: jormungandr/src/blockchain/process.rs (hunk 1 above)
Comment (NicolasDP) [reply jormungandr/src/blockchain/process.rs]: yes, thanks, I need to talk with @enzoc4 about it.

**Pull request merged**